ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/RaoUtil.java

```diff
@@ -64,10 +64,8 @@ public static void checkParameters(RaoParameters raoParameters, RaoInput raoInpu
             if (raoInput.getGlskProvider() == null) {
                 throw new FaraoException("Relative margin objective function requires a GLSK provider.");
             }
-            if (Objects.isNull(raoParameters.getExtension(RaoPtdfParameters.class))
-                || Objects.isNull(raoParameters.getExtension(RaoPtdfParameters.class).getBoundaries())
-                || raoParameters.getExtension(RaoPtdfParameters.class).getBoundaries().isEmpty()) {
-                throw new FaraoException("Relative margin objective function requires a list of pairs of country boundaries.");
+            if (raoParameters.getPtdfBoundaries().isEmpty()) {
+                throw new FaraoException("Relative margin objective function requires a list of pairs of country boundaries to compute PTDF sums upon.");
```

Comment:
Below that there is 
`if ((raoParameters.isRaoWithLoopFlowLimitation()
            || raoParameters.getObjectiveFunction().doesRequirePtdf())
            && (raoInput.getReferenceProgram() == null)) {`
Is it normal for the doesRequirePtdf part ?

Reply:
it's to merge the case with the isRaoWithLoopFlowLimitation case, when requiring a refprog (if i understand the question well)